src/sqlfluff/core/linter/linter.py

```diff
@@ -0,0 +1,760 @@
+                linter_logger.debug(
+                    "Indent balance test failed for %r. Template indents will not be linted for this file.",
+                    templated_file.fname,
+                )
+                # Don't enable the templating blocks.
+                templating_blocks_indent = False
+                # Disable the linting of L003 on templated tokens.
+                config.set_value(["rules", "L003", "lint_templated_tokens"], False)
+
+        # The file will have been lexed without config, so check all indents
+        # are enabled.
+        new_tokens = []
+        for token in cast(Tuple[BaseSegment, ...], tokens):
+            if token.is_meta:
+                token = cast(MetaSegment, token)
+                if token.indent_val != 0:
+                    # Don't allow it if we're not linting templating block indents.
+                    if not templating_blocks_indent:
+                        continue
+            new_tokens.append(token)
+        # Return new buffer
+        return new_tokens, violations, config
+
+    @staticmethod
+    def _parse_tokens(
+        tokens: Sequence[BaseSegment], config: FluffConfig, recurse: bool = True
+    ) -> Tuple[Optional[BaseSegment], List[SQLParseError]]:
+        parser = Parser(config=config)
+        violations = []
+        # Parse the file and log any problems
+        try:
+            parsed: Optional[BaseSegment] = parser.parse(tokens, recurse=recurse)
+        except SQLParseError as err:
+            linter_logger.info("PARSING FAILED! : %s", err)
+            violations.append(err)
+            return None, violations
+
+        if parsed:
+            linter_logger.info("\n###\n#\n# {0}\n#\n###".format("Parsed Tree:"))
+            linter_logger.info("\n" + parsed.stringify())
+            # We may succeed parsing, but still have unparsable segments. Extract them here.
+            for unparsable in parsed.iter_unparsables():
+                # No exception has been raised explicitly, but we still create one here
+                # so that we can use the common interface
+                violations.append(
+                    SQLParseError(
+                        "Line {0[0]}, Position {0[1]}: Found unparsable section: {1!r}".format(
+                            unparsable.pos_marker.working_loc,
+                            unparsable.raw
+                            if len(unparsable.raw) < 40
+                            else unparsable.raw[:40] + "...",
+                        ),
+                        segment=unparsable,
+                    )
+                )
+                linter_logger.info("Found unparsable segment...")
+                linter_logger.info(unparsable.stringify())
+        return parsed, violations
+
+    @staticmethod
+    def _generate_short_fname(fname: Optional[str] = None):
+        # Handle nulls gracefully
+        if not fname:
+            return None
+        return fname.replace("\\", "/").split("/")[-1]

```

Comment:
Could this be replaced by something from a standard Python module, e.g. `os.path.basename()` or `pathlib.Path.name`?

Reply:
Yes - probably `os.path.basename()`. I'll do some testing first, but I think that's the best option.